_security-analytics/api-tools/threat-intel/threat-intel-source.md

```diff
@@ -0,0 +1,637 @@
+ | `source_config.last_refreshed_time`         | String (DateTime) | The date and time when the source was last refreshed. Example: `2024-07-24T23:01:19.338184Z`.                   |
+
+### Example request for IOC_UPLOAD type
+
+POST _plugins/_security_analytics/threat_intel/sources/
+```json
+{
+  "type": "IOC_UPLOAD",
+  "name": "my_custom_feed",
+  "format": "STIX2",
+  "description": "this is the description",
+  "store_type": "OS",
+  "enabled": "false",
+  "ioc_types": [
+    "hashes"
+  ],
+  "source": {
+    "ioc_upload": {
+      "file_name": "test",
+      "iocs": [
+        {
+          "id": "1",
+          "name": "uldzafothwgik",
+          "type": "hashes",
+          "value": "gof",
+          "severity": "thvvz",
+          "created": 1719519073,
+          "modified": 1719519073,
+          "description": "first one here",
+          "labels": [
+            "ik"
+          ],
+          "feed_id": "jl",
+          "spec_version": "gavvnespe",
+          "version": -4356924786557562654
+        },
+        {
+          "id": "2",
+          "name": "uldzafothwgik",
+          "type": "hashes",
+          "value": "example-has00001",
+          "severity": "thvvz",
+          "created": "2024-06-20T01:06:20.562008Z",
+          "modified": "2024-06-20T02:06:20.56201Z",
+          "description": "first one here",
+          "labels": [
+            "ik"
+          ],
+          "feed_id": "jl",
+          "spec_version": "gavvnespe",
+          "version": -4356924786557562654
+        }
+      ]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Example response for IOC_UPLOAD type
+
+```json
+{
+  "_id": "2c0u7JAB9IJUg27gcjUp",
+  "_version": 2,
+  "source_config": {
+    "name": "my_custom_feed",
+    "format": "STIX2",
+    "type": "IOC_UPLOAD",
+    "ioc_types": [
+      "hashes"
+    ],
+    "description": "this is the description",
+    "created_by_user": null,
+    "created_at": "2024-07-25T23:16:25.257697Z",
+    "source": {
+      "ioc_upload": {
+        "file_name": "test",
+        "iocs": []
+      }
+    },
+    "enabled": false,
+    "enabled_time": null,
+    "last_update_time": "2024-07-25T23:16:26.011774Z",
+    "schedule": null,
+    "state": "AVAILABLE",
+    "refresh_type": "FULL",
+    "last_refreshed_user": null,
+    "last_refreshed_time": "2024-07-25T23:16:25.522735Z"
+  }
+}
+```
+
+### Example request for S3_CUSTOM type source
+
+POST _plugins/_security_analytics/threat_intel/sources/
+```json
+{
+ "type": "S3_CUSTOM",
+ "name": "example-ipv4-from-SAP-account",
+ "format": "STIX2",
+ "store_type": "OS",
+ "enabled": "true",
+ "schedule": {
+  "interval": {
+   "start_time": 1717097122,
+   "period": "10",
+   "unit": "DAYS"
+  }
+ },
+ "source": {
+  "s3": {
+   "bucket_name": "threat-intel-s3-test-bucket",
+   "object_key": "alltypess3object",
+   "region": "us-west-2",
+   "role_arn": "arn:aws:iam::248279774929:role/threat_intel_s3_test_role"
```

Comment:
Should we instead use a placeholder here like `<ROLE_ARN>`?

Reply:
Good point. It's a dummy aws account but  will let @Naarcha-AWS and @hdhalter recommned the right practice.